doc/7/getting-started/.react-native/screens/Chat/ChatClient.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import ChatView from './ChatView';
+import kuzzle from '../../services/kuzzle';
+
+class ChatClient extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            messages: []
+        };
+        this.kuzzle = kuzzle;
+        this.initConnection();
+        this.handleSendMessage = this.onSendMessage.bind(this);
+    }
+
+    async initConnection() {
+        // Etablish the connection
+        await kuzzle.connect();
+        // Check if 'chat' index exists
+        if (! await kuzzle.index.exists("chat")) {
+            // If not, create 'chat' index and 'messages' collection
+            await kuzzle.index.create("chat");
+            await kuzzle.collection.create("chat", "messages");
+        }
+        await this.fetchMessages();
+        await this.subscribeMessages();
+    }
+    displayDate(previousDate, currentDate) { 
+        if (previousDate === null) {// Message is the first of the array so need to display the date
+            return true;
+        }
+        const d1 = new Date(previousDate).toDateString()
+        const d2 = new Date(currentDate).toDateString()
+        if (d1 !== d2) {
+             // Previous message and current has different dates so need to display the date
+             return true
+        };
+        // Previous message and current has same dates so doesn't need to display the date
+        return false;
+    }
+    getMessage(hit, idx, array) {
```

Comment:
Nitpicking: poor variable name choice, what is `hit` ? and `array` is an array of what ?